### ISSUE TITLE
Bump JUnit 5 dependencies

### ISSUE THREAD

**Issue by chalup** [mannodermaus/android-junit5#96]
The JUnit 5 dependencies listed in `gradle.properties` are quite outdated. Can you safely bump them to the latest releases (1.2.0 for the platform, 5.2.0 for the jupiter & vintage engines)?

**Comment by mannodermaus** [comment]
Last I checked, no, because all versions from `5.1.0-M1` onwards are shipped as an MRJAR, and Android's dexer can't deal with the optional Java 9 stuff. [I raised this to the team a while back](https://issuetracker.google.com/issues/70150896), but nothing has come out of it so far, unfortunately.

If you like, you can try raising the embedded versions yourself through the DSL, and check if it maybe *does* work now. I haven't checked compatibility against the very latest tooling yet, so maybe there was a fix that was introduced recently.

**Comment by chalup** [comment]
Hmm, I've bumped the dependencies with:

```
android.testOptions.junitPlatform {
    jupiterVersion '5.2.0'
    platformVersion '1.2.0'
    vintageVersion '5.2.0'
}
```

and everything seems to work, both in IDE and in console.

**Comment by mannodermaus** [comment]
Okay, thanks for your test. I'll conduct some more tests, maybe something about the JUnit 5 packaging changed again in the meantime to make it work, or the AGP just silently added MRJAR support. For reference, what version of the AGP are you using?

**Comment by chalup** [comment]
I'm using AGP 3.1.2

**Comment by mannodermaus** [comment]
Thanks! I'll investigate this once I find some time, so we can finally bump this library up to 5.2 if possible.

**Comment by mannodermaus** [comment]
I've just released the first SNAPSHOT for version 1.2.0 of the library. Can you pull that and test it out i your project? The dexer issue I mentioned above only affects instrumentation tests, which won't run with JUnit 5.2.0 and AGP 3.1, which is why the new android-junit5 version has to raise the minimum AGP requirement as well.
